overlord/snapstate/backend/setup.go

```diff
@@ -77,15 +76,10 @@ func (b Backend) RemoveSnapFiles(s snap.PlaceInfo, meter progress.Meter) error {
 
 	snapPath := s.MountFile()
 
-	if osutil.FileExists(snapPath) {
-		// XXX: would be better if this information was in the state directly
-		info, _, err := OpenSnapFile(snapPath, nil)
-		if err != nil {
-			return fmt.Errorf("cannot read snap file %q to determine snap type: %v", snapPath, err)
-		}
-
+	// snapPath may either be a file or a (broken) symlink to a dir
+	if _, err := os.Lstat(snapPath); err == nil {
 		// remove the kernel assets (if any)
-		if info.Type == snap.TypeKernel {
+		if typ == snap.TypeKernel {
```

Comment:
This is probably not tested much, but the situation is not being made any worse, so sounds fine.